src/osd/OSD.cc

```diff
@@ -7949,6 +7949,13 @@ void OSD::handle_osd_map(MOSDMap *m)
 	delete o;
 	request_full_map(e, last);
 	last = e - 1;
+
+	// don't continue committing if we failed to enc the first inc map
+	if (last < start) {
+	  dout(10) << __func__ << " bailing because last < start (" << last << "<" << start << ")" << dendl;
+	  m->put();
+	  return;
```

Comment:
What, if any, are the follow on affects caused by just a return here for the caller?

Reply:
IIUC it skips the redundancies noted in this related ticket: https://tracker.ceph.com/issues/45670
So in fact it should be a perf increase while a cluster is mid-upgrade.
I hope qa with the injected crc error to catch anything else unexpected.